lua/entities/gmod_wire_egp/init.lua

```diff
@@ -33,12 +33,12 @@ end
 
 function ENT:SetEGPOwner( ply )
 	self.ply = ply
-	self.plyID = IsValid(ply) and ply:UniqueID() or "World"
+	self.plyID = IsValid(ply) and ply:UserID() or -1
```

Comment:
> Are UserIDs ever reused? The Garry's Mod Wiki doesn't explicitly mention it. If they are then this is a potential security issue.

It is not reusable. `UserID` is an increasing number for each new player connected to the server. Thus, it is not a security issue, but a nice improvement.

> If they don't get reused on rejoining then using AccountID here is maybe an even better idea as we'd then automatically handle rejoining.

What for?
Player's EGP is removed in moment after a player leaves server.
There may be a problem for 3rd-party addon relying on this particular value. In which case, I could switch it into `AccountID`, but that'd degrade a performance improvement in other place i.e. [here](https://github.com/wiremod/wire/blob/6fb3c27fbc4c5fac8ef1c713d8d62381ab94dbe6/lua/entities/gmod_wire_egp/init.lua#L41), because it will have to search for a specific player.

Reply:
> What for?
Player's EGP is removed in moment after a player leaves server.

Not necessarily, that depends on the server settings. See the title of 4f995e964b522c8fc08e56f023916f0d7548d4e9 which first introduced this code:

>  [EGP] Fixed materials sometimes not working, and also fixed not being able to update an EGP screen after rejoin

So we *do* need to use something that'll be stable across reconnects, which `UserID` isn't.

Reply:
Okay.. I will fix this.